src/webui/webui.cpp

```diff
@@ -90,10 +94,17 @@ void WebUI::init()
 
         if (!m_httpServer->isListening()) {
             bool success = m_httpServer->listen(QHostAddress::Any, m_port);
-            if (success)
+            if (success) {
                 logger->addMessage(tr("Web UI: Now listening on port %1").arg(m_port));
-            else
-                logger->addMessage(tr("Web UI: Unable to bind to port %1").arg(m_port), Log::CRITICAL);
+            }
+            else {
+                const QString errorMsg = tr("Web UI: Unable to bind to port %1. %2").arg(m_port).arg(m_httpServer->errorString());
```

Comment:
I love old-shool UNIX style of error messages in programs. When messages talk to user. Inform them on where, what happened, what is the cause, and what is a solution to that.

```cpp
const QString errorMsg = tr("Web UI: Unable to bind to port %1. Port is probably used by another process. Use `qbittorrent-nox --webui-port=<port>` option to set other port. %2").arg(m_port).arg(m_httpServer->errorString());
```

Reply:
This message is probably better for writing to stderr, but it unacceptable as log message, IMO.

Reply:
Hmm.
As a DevOps I know logs can bare and gather all kinds of trash today. Like QT likes to spam hard to understand messages not regarding log messages. So they can bare that noteworthy tail.

You said and opened my eyes on one note, why old Unix style of reports gone now.

The right thing to do here is to:
* log+stdout err lvl message.
* log+stdout Notice lvl proposing solution.

Reply:
Speculating on the source of the error has no place in the log file. In my case, the "proposed solution" is actually totally wrong. My problem is that webui is using `QHostAddress::Any` which tries to bind to IPv6 which is disabled in my FreeBSD jail ( will be making a PR for that shortly). I get a `Protocol not supported` error from `errorString()` for that.

Such speculation is just noise in my case and at worse, misleading. Finally the standard errno message of `Address already in Use` that you'll get for a port binding error is more than enough to diagnose the issue.

At any rate, this PR has been approved. Could we merge it and consider any further improvements in separate PR's?

Reply:
(Look at QT messages - how log messages soudn't be done. QT are the one most noteworthy product that savage over industrial standards)

To obey logging standards:

* Solution part should be warn/notice lvl message: "Port is probably used by another process. Use `qbittorrent-nox --webui-port=<port>` option to set port.".
  Why warn/notice:
  * Warning lvl message definition: "May indicate that an error will occur if action is not taken."
    So we inform people with warning, that they should take action. That obeys definition in standard. And then right away error happens.
  * Notice lvl message definition: "Events that are unusual, but not error conditions.". Since it is error condition - it is debatable to use Notice, where Warning fits better.

* Error part: "Web UI: Unable to bind to port %1." is indeed an error:
  * Error lvl message definition: "Error conditions". Where definition of "error" is known to us and described in standard. It is condition, when looses core functionality and can't work and exits with error status code.
  * Since on Error is convenient to bind exit abstraction - Warn lvl message logically by conventions should be before error message, and Notice - right after Error message.

<sub>
More info about that, and what I mean under "standard", I wrote full sections in Arch Wiki year ago: https://wiki.archlinux.org/index.php/Systemd#Priority_level - So people would know and not forget.

DevOps'es rely on stdout/log messages, and that they are in the right lvl, and info of messages is understandable. That is why it is important topic to me. When DevOps should control 1000 of instances, and 100G of logs, they, and monitoring/log analizing systems, rely on that developers made good logs. It is not news to you.

But that is world-wide topic.

Once-more QT is known not to obey logging. Their logs are so fore harder to troubleshoot, and understand what happens. Definitions of lvl's bind messages to state of system: something happed/something probably going to happen if you would not solve this stuff. Since QT does not obey that - you don't know, it is notice, warn or err in the program.
</sub>

I am sure people would be happy to get multiuser. As I said, I am happy also, that you resolve this.

I just shared info and my thoughts.

Reply:
@Anton-Latukha 
Thanks for sharing the details! I think this particular case we have given enough information to the user, so lets just get this PR merged first!